systems/controllers/constrained_lqr_controller.cc

```diff
@@ -0,0 +1,141 @@
+#include "systems/controllers/constrained_lqr_controller.h"
+
+namespace dairlib {
+namespace systems {
+
+using std::make_unique;
+using std::unique_ptr;
+using Eigen::VectorXd;
+using Eigen::MatrixXd;
+using Eigen::HouseholderQR;
+using dairlib::multibody::ContactInfo;
+using dairlib::multibody::ContactToolkit;
+using drake::AutoDiffXd;
+using drake::AutoDiffVecXd;
+using drake::math::initializeAutoDiff;
+using drake::math::autoDiffToGradientMatrix;
+using drake::math::autoDiffToValueMatrix;
+using drake::systems::Context;
+using drake::systems::controllers::LinearQuadraticRegulator;
+using drake::systems::controllers::LinearQuadraticRegulatorResult;
+
+ConstrainedLQRController::ConstrainedLQRController(
+    const RigidBodyTree<double>& tree, VectorXd q, VectorXd u, VectorXd lambda,
+    MatrixXd Q, MatrixXd R, ContactInfo contact_info)
+    : tree_(tree),
+      contact_info_(contact_info),
+      contact_toolkit_(
```

Comment:
Question: why a unique_ptr?

Reply:
I need `tree` and `contact_info` to create the object (It doesn't have a default constructor). So I couldn't really define it as a private non-pointer variable.